File: solutions/elixir/01-yy2/code/lib/main.ex
defmodule CLI do
  def main(args) do
    {opts, _, _} = OptionParser.parse(args, strict: [prompt: :string], aliases: [p: :prompt])
    prompt = Keyword.get(opts, :prompt) || raise "error: -p flag is required"

    api_key = System.get_env("OPENROUTER_API_KEY") || raise "OPENROUTER_API_KEY is not set"
    base_url = System.get_env("OPENROUTER_BASE_URL") || "https://openrouter.ai/api/v1"

    response =
      Req.post!("#{base_url}/chat/completions",
        json: %{
          "model" => "anthropic/claude-haiku-4.5",
          "messages" => [%{"role" => "user", "content" => prompt}]
        },
        auth: {:bearer, api_key}
      )

    choices = response.body["choices"]

    if is_nil(choices) or choices == [] do
      raise "no choices in response"
    end

    IO.write(List.first(choices)["message"]["content"])
  end
end

File: solutions/r/01-yy2/code/app/main.R
library(httr2)
library(jsonlite)

args <- commandArgs(trailingOnly = TRUE)

# Parse -p flag
if (length(args) < 2 || args[1] != "-p") {
  stop("error: -p flag is required")
}

prompt <- args[2]

api_key <- Sys.getenv("OPENROUTER_API_KEY")
base_url <- Sys.getenv("OPENROUTER_BASE_URL", "https://openrouter.ai/api/v1")

if (api_key == "") {
  stop("OPENROUTER_API_KEY is not set")
}

body <- list(
  model = "anthropic/claude-haiku-4.5",
  messages = list(
    list(role = "user", content = prompt)
  )
)

response <- request(paste0(base_url, "/chat/completions")) |>
  req_headers(
    "Authorization" = paste("Bearer", api_key),
    "Content-Type" = "application/json"
  ) |>
  req_body_json(body) |>
  req_perform()

result <- resp_body_json(response)

if (length(result$choices) == 0) {
  stop("no choices in response")
}

cat(result$choices[[1]]$message$content)
